src/Agent.Worker/Worker.cs

```diff
@@ -161,6 +161,10 @@ private void AddUserSuppliedSecret(String secret)
         private void InitializeSecretMasker(Pipelines.AgentJobRequestMessage message)
         {
             Trace.Entering();
+            if(message.Variables.TryGetValue(Constants.Variables.Features.UseMaskingPerformanceEnhancements, out var UsePerformanceEnhancements))
+            {
+                HostContext.SecretMasker.UsePerformanceEnhancements = UsePerformanceEnhancements.Value == "true";
```

Comment:
Not sure if we have a GetBoolean helper for variables, if we do that would be prefereable.  otherwise, please use case insensitive compare e.g. OrdinalIgnoreCase

Reply:
@merlynomsft, resolved